File: Gherkin/IGherkinLine.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Gherkin
{
    public interface IGherkinLine
    {
        int LineNumber { get; }

        void Detach();

        int Indent { get; }
        bool IsEmpty();
        bool StartsWith(string text);
        bool StartsWithTitleKeyword(string text);
        string GetLineText(int indentToRemove = -1);
        string GetRestTrimmed(int length);
        IEnumerable<string> GetTags();
        IEnumerable<string> GetTableCells();
    }
}

File: src/main/java/gherkin/IGherkinLine.java
package gherkin;

import java.util.List;

public interface IGherkinLine {
    public Integer indent();

    public void detach();

    public String getLineText(int indentToRemove);

    public boolean isEmpty();

    boolean startsWith(String prefix);

    String getRestTrimmed(int length);

    List<GherkinLineSpan> getTags();

    boolean startsWithTitleKeyword(String keyword);

    List<GherkinLineSpan> getTableCells();
}
